torch/distributions/one_hot_categorical.py

```diff
@@ -53,11 +60,13 @@ def entropy(self):
         return self._categorical.entropy()
 
     def enumerate_support(self):
-        probs = self._categorical.probs
         n = self.event_shape[0]
-        if isinstance(probs, Variable):
-            values = Variable(torch.eye(n, out=probs.data.new(n, n)))
+        out = self._categorical.new((n, n))
+        if isinstance(out, Variable):
```

Comment:
@fritzo - let me know if there is a better way to handle this.

Reply:
I guess you could simplify a tiny bit
```py
values = self._categorical.new((n, n))
torch.eye(n, out=(values.data if isinstance(values, Variable) else values)
```
If `torch.eye(out=values)` really doesn't work for CUDA then we should ask Adam what to do.